storage-proofs/src/stacked/proof.rs

```diff
@@ -435,31 +473,51 @@ impl<'a, H: 'static + Hasher, G: 'static + Hasher> StackedDrg<'a, H, G> {
         let layers = layer_challenges.layers();
         assert!(layers > 0);
 
-        let (encodings, column_hashes, tree_d) = crossbeam::thread::scope(|s| {
+        // Generate all store configs that we need based on the
+        // cache_path in the specified config.
+        assert!(config.is_some()); // FIXME: make not an Option
+        let mut tree_d_config = config.clone().unwrap();
+        let mut tree_r_last_config = config.clone().unwrap();
+        let mut tree_c_config = config.clone().unwrap();
+
+        tree_c_config.id = format!("tree-c-{:?}", replica_id);
+        tree_r_last_config.id = format!("tree-r-last-{:?}", replica_id);
```

Comment:
I think to make these operation a bit clearer, there should be a another method like this

```rust
let tree_c_config = Config::from_config(&config, format!("tree-c-{:?}", replica_id));
```

instead of the manual clone and id replacement